src/js/control-bar/volume-menu-button.js

```diff
@@ -108,11 +137,39 @@ class VolumeMenuButton extends MenuButton {
    *
    * @method handleClick
    */
-  handleClick() {
+  handleClick(event) {
     MuteToggle.prototype.handleClick.call(this);
+    this.toggleMutedLabel();
     super.handleClick();
   }
 
+  toggleMutedLabel() {
+    if (this.player_.muted()) {
```

Comment:
the specific labels here probably need to be tweaked. Maybe just saying `Muted` and `Unmuted` is enough. (Localized, of course).